tests/FSharp.Compiler.UnitTests/PostInferenceChecksTests.fs

```diff
@@ -0,0 +1,84 @@
+namespace FSharp.Compiler.ComponentTests.Checking
+
+open System
+open Xunit
+open FSharp.Compiler.PostTypeCheckSemanticChecks.Limit
+
+module PostInferenceChecksTests =
+
+    [<Literal>]
+    let internal NUM_TESTS = 10_000
+
+    let genNat (rng : Random) : int =
+        let i = rng.Next ()
+        if i = Int32.MinValue then
+            0
+        else
+            abs i
+
+    let internal possibleLimitFlags = LimitFlags.GetValues<LimitFlags> ()
+
+    let internal genLimitFlags (rng : Random) : LimitFlags =
+        Seq.init (genNat rng % 10 + 1) (fun _ -> possibleLimitFlags[genNat rng % possibleLimitFlags.Length])
+        |> Seq.reduce (|||)
+
+    let internal generateLimit (rng : Random) : Limit =
+        { scope = genNat rng
+          flags = genLimitFlags rng }
+
+    let createSeed () =
+        DateTime.UtcNow.Ticks % int64 Int32.MaxValue
```

Comment:
Two questions: would you prefer these tests to be deterministic (i.e. would you prefer to fix a seed), and would you prefer me to port them over to FsCheck (which would get rid of some of the boilerplate in this file)?

Reply:
If the test fails on CI, I ( we) need to know what to do in order to make it fail locally as well.
So a fixed seed would be better personally.

FsCheck - I think both with and without is doable, since it already is in another test project.

Reply:
Since the tests already print out the seed and the count at which they failed, I think I'd prefer the randomised seed for more coverage.